### PR TITLE
JAPI-420 HpccRelation should declare an equals function

### PR DESCRIPTION
**Pull request by jpmcmu**
Signed-off-by: James McMullan <James.McMullan@lexisnexis.com>

### CODE REVIEW

**Comment by jpmcmu** [comment]
@rpastrana please review

File: DataAccess/src/main/java/org/hpccsystems/spark/datasource/HpccRelation.java [hunk 1]
```diff
@@ -162,4 +162,49 @@ public SQLContext sqlContext()
...
     }
+
+    public boolean equals(Object rhs)
+    {
+        if (rhs instanceof HpccRelation)
+        {
+            HpccRelation otherRelation = (HpccRelation) rhs;
+            HpccOptions otherOptions = otherRelation.options;
+            
+            boolean nameMatches = otherOptions.fileName.equals(options.fileName);
```
Comment (rpastrana) [inline DataAccess/src/main/java/org/hpccsystems/spark/datasource/HpccRelation.java:173]: Perhaps overkill, but does it make sense to implement HpccOptions.equals(Obj) and call it here?

**Review by rpastrana — COMMENTED** [review]
@jpmcmu looks good, one comment

**Comment by jpmcmu** [comment]
@rpastrana updated equals to include checking host

File: DataAccess/src/main/java/org/hpccsystems/spark/datasource/HpccRelation.java (hunk 1 above)
Comment (jpmcmu) [reply DataAccess/src/main/java/org/hpccsystems/spark/datasource/HpccRelation.java:173]: Good question. There are other options inside of HpccOptions that we wouldn't want to compare in this case so I think it is probably better to keep the comparison inside of the relation.

**Review by rpastrana — COMMENTED** [review]
@jpmcmu I might have missed it, but I didn't see the connectionInfo comparison

**Comment by jpmcmu** [comment]
@rpastrana updated

**Review by rpastrana — APPROVED** [review]

**Pull request merged**